fe/src/main/java/org/apache/doris/planner/OlapTableSink.java

```diff
@@ -293,7 +293,11 @@ private TOlapTableLocationParam createLocation(OlapTable table) throws UserExcep
         TOlapTableLocationParam locationParam = new TOlapTableLocationParam();
         // BE id -> path hash
         Multimap<Long, Long> allBePathsMap = HashMultimap.create();
-        for (Partition partition : table.getPartitions()) {
+        for (Partition partition : table.getAllPartitions()) {
+            if (!partitionIds.isEmpty() && !partitionIds.contains(partition.getId())) {
```

Comment:
Don't need to check again, we have checked in `init` method.

Reply:
I didn't get you point? no need to check what ?

Reply:
`if (!partitionIds.isEmpty() && !partitionIds.contains(partition.getId())) {`

Reply:
Here I use this `if (!partitionIds.isEmpty() && !partitionIds.contains(partition.getId()))` to filter all unrelated partitions. I didn't see any such check logic in `init` method?